src/utilities/color-manipulation.ts

```diff
@@ -19,7 +20,20 @@ export function darkenColor(color: HSLColor | string, lighten = 0) {
   const {lightness} = color;
   const nextLightness = lightness - lighten;
 
-  return {...color, lightness: nextLightness};
+  return {...color, lightness: clamp(nextLightness, 0, 100)};
```

Comment:
This fixes a bug where invalid hex colors could be returned if a lightness was outside the valid range.

It could use a regression test.